cpp/src/io/parquet/writer_impl.cu

```diff
@@ -534,6 +534,14 @@ std::vector<schema_tree_node> construct_schema_tree(LinkedColVector const& linke
     [&](LinkedColPtr const& col, column_in_metadata& col_meta, size_t parent_idx) {
       bool col_nullable = is_col_nullable(col, col_meta, single_write_mode);
 
+      auto set_field_id = [&schema, parent_idx](schema_tree_node& s,
+                                                column_in_metadata const& col_meta) {
+        if (schema[parent_idx].name != "list" and col_meta.is_parquet_field_id_set()) {
```

Comment:
why is the list check required?

Reply:
presumably because "stub" elements can't have a field id

Reply:
Similar to how we set schema names, my idea was to not set field ID for "intermediate" schemas for lists. Removing it now since it's not necessarily required.

Reply:
Based on offline discussion with @devavret, we agreed to keep the list check since "stub" elements are not supposed to have field ID.

Reply:
Let's confirm it as well. @jlowe , would spark have a `field_id` corresponding to a stub element? Or is it mostly used for the parent-most level?

Reply:
Based on the PR description in https://issues.apache.org/jira/browse/SPARK-38094:
```
After this PR, when the requested schema has field IDs, Parquet readers will first use the field ID to determine which Parquet columns to read, before falling back to using column names as before.
```
I assume it's mainly for the outter-most level.

Reply:
> would spark have a field_id corresponding to a stub element?

It is expecting to find a specified field ID on a child column of a StructType (STRUCT) column.  Theoretically there could be a STRUCT of LIST (array) column and the user could specify it on the array column.  See https://github.com/apache/spark/pull/35385/files#diff-487304e31da0dcde467c1f8561f42edcb3a811a755d8bc0424e4f3ad084099c3R156-R175

However I don't know whether the underlying Parquet message types allow a field ID on the array column vs. the child column of the array.